js/src/toast/toast.spec.js

```diff
@@ -0,0 +1,222 @@
+
+      spyOn(toastEl.classList, 'contains')
+
+      toast.hide()
+
+      expect(toastEl.classList.contains).toHaveBeenCalled()
+    })
+  })
+
+  describe('dispose', () => {
+    it('should allow to destroy toast', () => {
+      fixtureEl.innerHTML = '<div></div>'
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      expect(Toast._getInstance(toastEl)).toBeDefined()
+
+      toast.dispose()
+
+      expect(Toast._getInstance(toastEl)).toBeNull()
+    })
+
+    it('should allow to destroy toast and hide it before that', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast" data-delay="0" data-autohide="false">',
+        '  <div class="toast-body">',
+        '    a simple toast',
+        '  </div>',
+        '</div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      EventHandler.one(toastEl, 'shown.bs.toast', () => {
+        setTimeout(() => {
+          expect(toastEl.classList.contains('show')).toEqual(true)
+          expect(Toast._getInstance(toastEl)).toBeDefined()
+
+          toast.dispose()
+
+          expect(Toast._getInstance(toastEl)).toBeNull()
+          expect(toastEl.classList.contains('show')).toEqual(false)
+
+          done()
+        }, 1)
+      })
+      toast.show()
+    })
+  })
+
+  describe('config', () => {
+    it('should allow to config in js', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast">',
+        '  <div class="toast-body">',
+        '    a simple toast',
+        '  </div>',
+        '</div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl, {
+        delay: 1
+      })
+
+      toastEl.addEventListener('shown.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(true)
+        done()
+      })
+
+      toast.show()
+    })
+  })
+
+  describe('event-key', () => {
+    it('should close toast when close element with data-dismiss attribute is set', done => {
+      fixtureEl.innerHTML = [
+        '<div class="toast" data-delay="1" data-autohide="false" data-animation="false">',
+        '  <button type="button" class="ml-2 mb-1 close" data-dismiss="toast">',
+        '    close',
+        '  </button>',
+        '</div>'
+      ].join('')
+
+      const toastEl = fixtureEl.querySelector('div')
+      const toast = new Toast(toastEl)
+
+      toastEl.addEventListener('shown.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(true)
+
+        const button = toastEl.querySelector('.close')
+
+        button.click()
+      })
+
+      toastEl.addEventListener('hidden.bs.toast', () => {
+        expect(toastEl.classList.contains('show')).toEqual(false)
+        done()
+      })
+
+      toast.show()
+    })
+  })
+
+  describe('constructor', () => {
+    it('should expose default setting to allow to override them', () => {
```

Comment:
should be relative to the `Default` describe

Reply:
@Johann-S I'm not sure what you mean. Would you please elaborate?

Reply:
Default is a public method so it should be in a separate `describe`

Reply:
@Johann-S I made Default and DefaultType into their own describes ([two days ago](https://github.com/twbs/bootstrap/commit/299ad44fc60fbbde2e5fb3572ceaa5a8b10dffaf), I just did not mention it in the comment).
I hope my contributions are actually helpful and not wasting your time.

Reply:
Your contributions are totally valuable for us thanks for that 👍 It's just I like to follow the same rules for the structure of our unit tests 👍